### ISSUE TITLE
QueryBuilder produces invalid SQL from an empty CompositeExpression 

### ISSUE THREAD

**Issue by morozov** [doctrine/dbal#3845]
|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.10.1

```php
$qb = $conn->createQueryBuilder();
$qb->select('*')
    ->where($qb->expr()->andX());
echo $qb;
// SELECT * WHERE ()
```

**Comment by SenseException** [comment]
What's the expected behaviour for this?

**Comment by morozov** [comment]
Either a `QueryException` or no `WHERE` clause. I mostly filed it for documentation purposes. In `master`, I want to make the composite expression to always have at least one (or even better, two) parts. It shouldn't be possible to instantiate an empty object in the first place.

**Comment by morozov** [comment]
Fixed by #3868.
